### PR TITLE
DDS conversion between REFERENCE objects

### PR DESCRIPTION
**Pull request by IkerLuengo**
This PR implements the explicit/implicit conversion macros for REFERENCE types on PSM

### CODE REVIEW

File: include/dds/core/refmacros.hpp [hunk 1]
```diff
@@ -131,6 +132,7 @@ private: \
         } else { \
             /* TODO: Review this uses OpenSplice */ \
             /* this->::dds::core::Reference<DELEGATE_T>::impl_ = OSPL_CXX11_STD_MODULE::dynamic_pointer_cast<DELEGATE_T>(h.delegate()); */ \
```
Comment (laura-eprosima) [inline include/dds/core/refmacros.hpp:134]:
```suggestion

```

File: include/dds/core/refmacros.hpp [hunk 2]
```diff
@@ -78,6 +78,7 @@ public: \
         } else { \
             /* TODO: Review this uses OpenSplice */ \
             /*this->::dds::core::Reference< DELEGATE_T >::impl_ = OSPL_CXX11_STD_MODULE::dynamic_pointer_cast< DELEGATE_T >(h.delegate()); */ \
```
Comment (laura-eprosima) [inline include/dds/core/refmacros.hpp:80]:
```suggestion

```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1731/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1488/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1723/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1734/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1491/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1726/)

**Pull request merged**